src/components/BottomSheet/BottomSheet.tsx

```diff
@@ -0,0 +1,164 @@
+import {
+  BackHandler,
+  Dimensions,
+  StyleSheet,
+  Text,
+  TouchableOpacity,
+  TouchableWithoutFeedback,
+  View,
+} from 'react-native';
+import React from 'react';
+import {useRecoilState, useRecoilValue} from 'recoil';
+import {
+  bottomSheetContent,
+  bottomSheetVisibleState,
+  useBottomSheet,
+} from '../../recoil/BottomSheetStore';
+import Animated, {
+  BounceIn,
+  BounceInUp,
+  BounceOut,
+  EntryExitTransition,
+  FadeIn,
+  FadeOut,
+  JumpingTransition,
+  runOnJS,
+  useAnimatedStyle,
+  useSharedValue,
+  withTiming,
+} from 'react-native-reanimated';
+import {Gesture, GestureDetector} from 'react-native-gesture-handler';
+
+const screenHeight = Dimensions.get('window').height;
+const bottomSheetHeight = screenHeight * 0.4;
+
+const BottomSheet = () => {
+  const content = useRecoilValue(bottomSheetContent);
+  const [isVisible, setIsVisible] = useRecoilState(bottomSheetVisibleState);
+
+  const [isOn, setIsOn] = React.useState<boolean>(false);
+
+  // const bottom = useSharedValue(0);
+  const top = useSharedValue(screenHeight);
+  const containerHeight = useSharedValue(bottomSheetHeight);
+
+  const {hideBottomSheet} = useBottomSheet();
+
+  const handleBackgroundClick = () => {
+    hideBottomSheet();
+  };
+
+  const sheetAnimatedStyle = useAnimatedStyle(() => {
+    return {
+      top: top.value,
+      height: containerHeight.value,
+    };
+  });
+
+  const startY = useSharedValue(0);
+  const panGestureEvent = Gesture.Pan()
+    .onStart(() => {
+      // 처음의 값 저장
+      startY.value = containerHeight.value;
+    })
+    .onUpdate(event => {
+      // 찍어보기
+      console.log(event.translationY);
+
+      // let tempValue = top.value + event.translationY;
+      // let temp = containerHeight.value - event.translationY;
+      // top.value = withTiming(tempValue);
+      // containerHeight.value = withTiming(startY.value + -event.translationY);
+      containerHeight.value = withTiming(startY.value + -event.translationY);
+    });
+
+  React.useEffect(() => {
+    if (isVisible.isBottomSheetVisible) {
+      setIsOn(true);
+      top.value = withTiming(screenHeight - bottomSheetHeight, {
+        duration: 400,
+      });
+    } else {
+      top.value = withTiming(screenHeight, {
+        duration: 400,
+      });
+      const timeout = setTimeout(() => {
+        setIsOn(false);
+      }, 400);
+      return () => clearTimeout(timeout);
```

Comment:
```suggestion
      top.value = withTiming(screenHeight, {
        duration: 400,
      }, () => { runOnJS(setIsOn)(false) });
```

이게 더 깔끔하고 안정적이지 않을까요??